### DISCUSSION TITLE
New dev version, upgrade to new IDF environment

### DISCUSSION THREAD

**Discussion by dl9rdz** [dl9rdz/rdz_ttgo_sonde#448] [Announcements]
I have started migration to a more recent toolchain / IDF version. The one in used has been getting rather old and outdated. This has a number of implications:
- The binary images are now called differently (mainXXX and devXXX vs masterXXX and develXXX). You cannot do OTA upgrades between the old and the new versions. OTA will be supported as before within newer releases of the new version.
- The Flash file system changed from deprecated SPIFFS to the more modern LittleFS. This is one reason why supporting an OTA update is not easily feasible with that version change.
- The new images will not work on very old ESP boards with 26 MHz crystal. Probably no one is using them anyway, so this should not be an issue. If needed, you can manually change the build config to 26 Mhz and recompile yourself.
- In the old versions, the autoconfig for the popular TTGO Lora32 v2 erroneously configured GPIO16 as OLED Reset. This is wrong but just by chance did not cause problems in the past. On the new build, this config may cause a lock up on startup, ending in a watchdog reboot loop (as GPIO16 is connected to the Flash memory and possibly interferes). New autoconfig should get this right, but **if you copy/paste your previous config, make sure to change OLED RST to -1.**
- I used this transition to implement a long-planned change: offload the fonts to a separate flash partition. 
- The primary build environment is now PlatformIO. You can upload fonts, code, and files separately with the pio targets upload, uploadfs, and uploadfonts **(You need to do all three at least once)**. the pio target "firmware" will build a full firmware image (to be flashed at offset 0x1000) that includes all parts.
- You should still be able to build the software with ArduinoIDE, but I have not tested this, and probably you will have to find out how to upload file system and font partition manually (you can disable the separate font partition in features.h)

This is an early experimental test release. Feel free to test it (happy about any feedback), but if you prefer a stable system, better stick to the older version (make sure to use the most recent devel or master version, not anything older than last February!).

**Comment by d2f2** [comment]
dev20241229-C3
Crashs with symbols on screen at scanning with LILYGO TtGo lora.

Usualy crash twice after power UP.  The works some time. The crash randomly.

**Reply by hr-ru** [reply]
Can you provide more details? I cannot reproduce this here.
Can you provide serial log file of the crash?

**Reply by hr-ru** [reply]
You can also test if 20250108 resolves this issue.

(There was some webserial debug code that accidentally went in the firmware image. If the ttgo receives IMPROV messages on the serial port during startup - that may happen when you use the webserial installer - it may access the network before initialization has completed, potentially causing a crash. I don't think this is your issue. As the web installer by default installed an older version, not 20241229, the only way to reproduce this error is to manually install dev20241229 and then use the web serial console in Chrome to see the serial log file.).

**Reply by d2f2** [reply]
20250108 seems to resolve this issue.
Ill try serial console to log.
